pkg/microservice/aslan/core/environment/service/environment.go

```diff
@@ -907,6 +939,44 @@ func UpdateHelmProduct(productName, envName, updateType, username, requestID str
 		log.Errorf("[%s][P:%s] GetProductTemplate error: %v", envName, productName, err)
 		return e.ErrUpdateEnv.AddDesc(e.FindProductTmplErrMsg)
 	}
+
+	for _, svrs := range updateProd.Services {
+		for _, svr := range svrs {
+			for _, rc := range renderSet.ChartInfos {
+				if rc.ServiceName == svr.ServiceName {
+					for _, c := range svr.Containers {
+						image := c.Image
```

Comment:
nested loop is hard to read

Reply:
The outer loop needs to be set; And these loops require less code; It doesn't feel like it's necessary